### ISSUE TITLE
[Bug]: Not nullable with default are required

### ISSUE THREAD

**Issue by alxvgt** [nelmio/NelmioApiDocBundle#2304]
### Version

4.27.0

### Description

During our recent update from 4.26.2 to 4.27.0 some of our properties are now mark as required but i'm not sure if it should be the case.

Ex : 
```php
class Test
{
    /**
     * @OA\Property(type="array", @OA\Items(ref=@Model(type=Toto::class)))
     */
    protected array $options = [];

    protected ?string $callback = null;
 ```
 In this case : 
 - `options` is set as required
 - `callback` is set as not required

Both has default value that led to not require any value i guess ?

**Comment by alxvgt** [comment]
This MR is the source of then change : https://github.com/nelmio/NelmioApiDocBundle/pull/2103

**Comment by deluxetom** [comment]
I'm having the same issue, I think that PR should be reverted
https://github.com/nelmio/NelmioApiDocBundle/pull/2103#issuecomment-2223476879

**Comment by DjordyKoert** [comment]
https://github.com/nelmio/NelmioApiDocBundle/pull/2103 has been reverted and is available in [v4.29.0](https://github.com/nelmio/NelmioApiDocBundle/releases/tag/v4.29.0)
